### DISCUSSION TITLE
Document how to do forms, error pages

### DISCUSSION THREAD

**Discussion by 1EDExg0ffyXfTEqdIUAYNZGnCeajIxMWd2vaQeP** [zoontek/chicane#29] [Q&A] [answered]
I can't find it

**Comment by zoontek** [comment, marked as answer]
Hi 👋,

There's nothing specific to handle forms or errors, chicane is idiomatic React.

Just use forms:
- https://reactjs.org/docs/forms.html

Or with the library of your choice:
- https://github.com/swan-io/react-ux-form
- https://formik.org
- https://react-hook-form.com

For errors, showing a "not found" page is easy:

```tsx
const App: FC = async () => {
  const route = router.useRoute([/* … */]);

  switch (route.name) {
    case "Root":
      return <Root />;
    // …
    default:
      return <NotFound />; // render "not found" when no route match
  }
};
```

Handling other errors is also just idiomatic React:

- https://reactjs.org/docs/error-boundaries.html
- https://github.com/bvaughn/react-error-boundary

I'm aware a lot of routers tends to make a lot of coupling with other concepts (form handling, data fetching, etc) lately, but the purpose of this library is not to trap you into an ecosystem 😄

Chicane does routing, safely. Doesn't sell you something else.

**Comment by 1EDExg0ffyXfTEqdIUAYNZGnCeajIxMWd2vaQeP** [comment]
Thanks. I think it would be a good idea to have a port of https://reactrouter.com/en/main/start/tutorial to get a feel for what chicane does and how to express everything else in idiomatic react. I think it would be useful to people coming from react-router and trying to port their app.

**Reply by zoontek** [reply]
Maybe, yes. But it would be a lot shorter 😅
I think React Router became way too complex lately since Remix didn't really took off and they "remixed" RR to put back nearly all of Remix in it (aka, a fully featured framework).
